obspy/core/stream.py

```diff
@@ -3118,6 +3119,85 @@ def remove_sensitivity(self, *args, **kwargs):
+        :param type: Type of stack, one of the following:
+            `'linear'`: average stack (default),
+            `('pw', order)`: phase weighted stack of given order,
+            see Schimmel and Paulssen (1997),
+            `('root', order)`: root stack of given order.
+        :param npts_tol: Tolerate traces with different number of points
+            with a difference up to this value. Surplus samples are discarded.
+
+        :returns: New stream object with stacked traces. The metadata of each
+            trace corresponds to the metadata of the original traces
+            if those are the same. Additionaly, the entries
+            `stack` (result of the format operation on the group parameter) and
+            `stack_count` (number of stacked traces)
+            are written to the stats object.
+
+        >>> from obspy import read
+        >>> st = read()
+        >>> stack = st.stack()
+        >>> print(stack)
+        1 Trace(s) in Stream:
+        BW.RJOB.. | 2009-08-24T00:20:03.000000Z - ... | 100.0 Hz, 3000 samples
+        """
+        if len(type) == 2:
+            type, order = type
+        groups = collections.defaultdict(list)
+        if group == 'seedid':
+            group = '{network}.{station}.{location}.{channel}'
+        for tr in self:
+            groups[group.format(**tr.stats)].append(tr)
+        stacks = []
+        for groupid, grouptrcs in groups.items():
+            header = {k: v for k, v in grouptrcs[0].stats.items()
+                      if all(tr.stats.get(k) == v for tr in grouptrcs)}
+            header['stack'] = groupid
+            header['stack_count'] = len(grouptrcs)
+            npts_all = [len(tr) for tr in grouptrcs]
+            npts_dif = max(npts_all) - min(npts_all)
+            npts = min(npts_all)
+            if npts_dif > npts_tol:
+                msg = ('Difference of number of points of the traces is higher'
+                       ' than requested tolerance ({} > {})')
+                raise ValueError(msg.format(npts_dif, npts_tol))
+            elif npts_dif > 0:
+                grouptrcs = [copy.copy(tr) for tr in grouptrcs]
+                for tr in grouptrcs:
+                    tr.data = tr.data[:npts]
+            data = np.array([tr.data for tr in grouptrcs])
+            if type == 'normal':
+                stack = np.mean(data, axis=0)
+            elif type == 'pw':
+                from scipy.signal import hilbert
+                from scipy.fftpack import next_fast_len
+                nfft = next_fast_len(npts)
+                anal_sig = hilbert(data, N=nfft)[:, :npts]
+                norm_anal_sig = anal_sig / np.abs(anal_sig)
+                phase_stack = np.abs(np.mean(norm_anal_sig, axis=0)) ** order
+                stack = np.mean(data, axis=0) * phase_stack
+            elif type == 'root':
+                r = np.mean(np.sign(data) * np.abs(data)
+                            ** (1 / order), axis=0)
```

Comment:
is ``1/order`` safe on all python versions?

Reply:
I think its not. I am used to py3, now.

Reply:
me to and 1.2.0 will be "py3 only", but in case of...

Reply:
Probably ;)

Reply:
Actually, there is this: `from __future__ import division`